### DISCUSSION TITLE
使用st表格每一行都带有列名

### DISCUSSION THREAD

**Discussion by sy-yan** [ng-alain/ng-alain#2312] [Q&A] [answered]
I Use st in my project。but when i use st.addRow() ,it add the column's name as a prefix.for example ,i want "a" in clomn "b"，but i got "ba".
I tried the example code in ng-alain.com,but it got a wrong result either.
我按官网的代码复制了一份，结果我用addRow()，他把列名也加上了，本来要'123'的，他展示成'id123'，我的data明明是[{id:"123"}]。
为什么一直出错呢？
@delon版本14.0.0 @angular 14.2.3

**Comment by cipchk** [comment, marked as answer]
能否提供一个重现链接？

**Reply by sy-yan** [reply]
import { Component } from '@angular/core';

import { STChange, STClickRowClassNameType, STColumn } from '@delon/abc/st';

@Component({
  selector: 'components-st-row-op',
  template: ` <button nz-button (click)="st.addRow({ id: 1000, email: 'add@email.com', phone: '123' })"> addRow</button>
    <button nz-button (click)="st.removeRow(0)"> removeRow index: 0 </button>
    <button nz-button (click)="st.setRow(0, { className: 'text-success' })"> Via setRow method </button>
    <st
      #st
      [data]="url"
      [req]="{ params: params }"
      [columns]="columns"
      (change)="_click($event)"
      [clickRowClassName]="clickRowClassName"
    ></st>`
})
export class ApplyComponent {
  url = [];
  params = { a: 1, b: 2 };
  clickRowClassName: STClickRowClassNameType = { exclusive: true, fn: () => 'text-error' };
  columns: STColumn[] = [
    { title: '编号', index: 'id' },
    { title: '邮箱', index: 'email' },
    { title: '电话', index: 'phone' }
  ];

  _click(e: STChange): void {
    console.log('click', e);
  }
}
没有链接，只有代码，

**Reply by cipchk** [reply]
检查一下以下几个文件是否同步，如果没有参照修改一下：

- https://github.com/ng-alain/ng-alain/blob/master/src/styles.less
- https://github.com/ng-alain/ng-alain/blob/master/src/styles/theme.less

除此之外，看一下 `angular.json` 的 [stylePreprocessorOptions](https://github.com/ng-alain/ng-alain/blob/master/angular.json#L43) 配置。

如果问题依旧，尝试移除 `node_modules` 和 `package-lock.json` 、`yarn.lock` 重新安装依赖。

还是如此，那我也不清楚了。

**Reply by sy-yan** [reply]
原来如此，原来是因为样式没有导入，所以找不到样式他span就直接把列名展示出来了。受教了。
